pkg/imagepolicy/kyverno/engine.go

```diff
@@ -0,0 +1,91 @@
+package kyverno
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"github.com/sealerio/sealer/common"
+	imagecommon "github.com/sealerio/sealer/pkg/define/options"
+	"github.com/sealerio/sealer/pkg/imageengine"
+	"github.com/sealerio/sealer/pkg/infradriver"
+	k "github.com/sealerio/sealer/pkg/runtime/kubernetes"
+	apimachineryRuntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+type Engine struct {
+}
+
+func NewKyvernoImagePolicyEngine() (*Engine, error) {
+	return &Engine{}, nil
+}
+
+func (engine *Engine) IsImagePolicyApp(appName string) (bool, error) {
+	return appName == "kyverno", nil
+}
+
+func (engine *Engine) CreateImagePolicyRule(infraDriver infradriver.InfraDriver, imageEngine imageengine.Interface, appName string) error {
+	driver, err := k.NewKubeDriver(k.AdminKubeConfPath)
+	if err != nil {
+		return err
+	}
+
+	imagePolicyTemplateStr := common.ImagePolicyTemplate
+	templatePath := fmt.Sprintf("%s/application/apps/%s/%s", infraDriver.GetClusterRootfsPath(), appName, common.ImagePolicyTemplateYamlName)
+	if _, err := os.Stat(templatePath); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+	} else {
+		yamlFile, err := ioutil.ReadFile(templatePath)
+		if err != nil {
+			return err
+		}
+		imagePolicyTemplateStr = string(yamlFile)
+	}
+
+	clusterImageName := infraDriver.GetClusterImageName()
+	imageList, err := imageEngine.GetSealerContainerImageList(&imagecommon.GetImageAnnoOptions{ImageNameOrID: clusterImageName})
+	clusterPolicy := &kyvernov1.ClusterPolicy{}
+
+	var imageListArray []string
+	for _, containerImage := range imageList {
+		imageListArray = append(imageListArray, "\""+containerImage.Image+"\"")
+	}
+
+	ctx := map[string]string{
+		"name":      clusterImageName,
+		"imageList": "[" + strings.Join(imageListArray, ",") + "]",
+		"registry":  infraDriver.GetClusterRegistry().LocalRegistry.Domain,
+	}
+	imagePolicyTemplate, err := SubsituteTemplate(imagePolicyTemplateStr, ctx)
+	if err != nil {
+		return err
+	}
+
+	kyvernov1.AddToScheme(scheme.Scheme)
+	decoder := serializer.NewCodecFactory(scheme.Scheme).UniversalDecoder()
+	err = apimachineryRuntime.DecodeInto(decoder, []byte(imagePolicyTemplate), clusterPolicy)
+	if err != nil {
```

Comment:
if err:=apimachineryRuntime.DecodeInto();err!=nil{
}

the uppper format will reduces the scope of error